modules/oe_theme_inpage_navigation/tests/src/FunctionalJavascript/InPageNavigationBlockTest.php

```diff
@@ -0,0 +1,302 @@
+
+use Drupal\block\Entity\Block;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\media\MediaInterface;
+use Drupal\node\Entity\NodeType;
+use Drupal\oe_theme_inpage_navigation\InPageNavigationHelper;
+use Drupal\Tests\oe_theme\PatternAssertions\InPageNavigationAssert;
+
+/**
+ * Test Inpage navigation block plugin.
+ */
+class InPageNavigationBlockTest extends WebDriverTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'block',
+    'oe_theme_helper',
+    'oe_theme_content_page',
+    'oe_theme_content_publication',
+    'oe_theme_content_organisation',
+    'oe_theme_content_organisation_reference',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Enable and set OpenEuropa Theme as default.
+    $this->container->get('theme_installer')->install(['oe_theme']);
+    $this->container->get('theme_handler')->setDefault('oe_theme');
+    $this->container->set('theme.registry', NULL);
+
+    // Enable test module with placed block in new region.
+    $this->container->get('module_installer')
+      ->install(['oe_theme_inpage_navigation_test']);
+
+    // Rebuild the ui_pattern definitions to collect the ones provided by
+    // oe_theme itself.
+    \Drupal::service('plugin.manager.ui_patterns')->clearCachedDefinitions();
+
+    \Drupal::service('emr.installer')->installEntityMetaTypeOnContentEntityType('oe_theme_inpage_navigation', 'node', ['oe_publication', 'oe_page']);
+
+    // Enable inpage_navigation for Publication content.
+    $ct_with = NodeType::load('oe_publication');
+    $ct_with->setThirdPartySetting('oe_theme_inpage_navigation', 'enabled', TRUE);
+    $ct_with->save();
+  }
+
+  /**
+   * Test content with enabled legacy in-page navigation by default.
+   */
+  public function testContentWithInPageNav(): void {
+
+    // Create a document for Publication.
+    $media_document = $this->createMediaDocument('publication_document');
+
+    // Create a Publication node with required fields only.
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->container->get('entity_type.manager')
+      ->getStorage('node')
+      ->create([
+        'type' => 'oe_publication',
+        'title' => 'Test Publication node',
+        'oe_publication_type' => 'http://publications.europa.eu/resource/authority/resource-type/ABSTRACT_JUR',
+        'oe_documents' => [$media_document],
+        'oe_publication_date' => [
+          'value' => '2020-04-15',
+        ],
+        'oe_subject' => 'http://data.europa.eu/uxp/1000',
+        'oe_author' => 'http://publications.europa.eu/resource/authority/corporate-body/AASM',
+        'oe_content_content_owner' => 'http://publications.europa.eu/resource/authority/corporate-body/COMMU',
+        'uid' => 0,
+        'status' => 1,
+      ]);
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    // Assert content part.
```

Comment:
Let's improve these comments: Assert that the main page content block is marked as source area.
But I don't understand what we are testing here, the article? Or the data attribute.

Reply:
I've removed this as it doesn't bring big value.